Lib/test/datetimetester.py

```diff
@@ -301,6 +301,25 @@ def test_inheritance(self):
         self.assertIsInstance(timezone.utc, tzinfo)
         self.assertIsInstance(self.EST, tzinfo)
 
+    def test_cannot_subclass_deprecation_warning(self):
+        if '_Fast' in self.__class__.__name__:
+            self.skipTest('Only run for Pure Python implementation')
+
+        msg = (
+            "Subclassing 'datetime.timezone' is deprecated and scheduled for removal "
+            "in Python 3.15."
+        )
+        with self.assertWarnsRegex(DeprecationWarning, msg):
+            class MyTimezone(timezone): pass
+
+    def test_cannot_subclass(self):
+        if '_Pure' in self.__class__.__name__:
+            self.skipTest('Only run for Fast C implementation')
+
+        msg = "type 'datetime.timezone' is not an acceptable base type"
+        with self.assertRaisesRegex(TypeError, msg):
```

Comment:
I am usually inclined to use `assertRaises` instead of `assertRaisesRegex`, since I don't think the specific wording of the exception message is part of the public interface. In this case, "we" (as in `datetime`) don't even own this wording.

Reply:
Changed.